test/core/networking/websockets/session_test.dart

```diff
@@ -0,0 +1,408 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:test/test.dart';
+
+import 'package:flyde/core/networking/websockets/session.dart';
+import 'package:flyde/core/networking/websockets/middleware.dart';
+
+import '../../../helpers/value_hook.dart';
+import '../../../helpers/wait_for.dart';
+
+//? The WebSocket uses in this file are not race conditions,
+//? because of how the dart eventloop is implemented.
+//? The internal WebSocket connect is !scheduled! and executed when the main strain waits (await)
+
+void main() {
+  late HttpServer? server;
+  late Uri? url;
+
+  setUp(() async {
+    server = await HttpServer.bind('localhost', 0);
+    url = Uri.parse('ws://${server!.address.host}:${server!.port}');
+  });
+
+  tearDown(() async {
+    await server!.close(force: true);
+    server = null;
+    url = null;
+  });
+
+  //* Tests
```

Comment:
Not required

Reply:
No comment is _required_, it just makes it easier to read

Reply:
But you do not do this anywhere else. That's inconsistent. And is it really so much easier to read now? There are 12 lines of setup code before the tests start. That's not much.